x-pack/plugins/security_solution/server/lib/detection_engine/rules/bulk_edit_rules.ts

```diff
@@ -41,12 +50,73 @@ export const bulkEditRules = ({
 }: BulkEditRulesArguments) => {
   const { attributesActions, paramsActions } = splitBulkEditActions(actions);
 
-  return rulesClient.bulkEdit({
+  const result = await rulesClient.bulkEdit({
     ...(ids ? { ids } : { filter: enrichFilterWithRuleTypeMapping(filter) }),
-    operations: attributesActions.map(bulkEditActionToRulesClientOperation),
+    operations: attributesActions.map(bulkEditActionToRulesClientOperation).flat(),
     paramsModifier: async (ruleParams: RuleAlertType['params']) => {
-      await validateBulkEditRule({ mlAuthz, ruleType: ruleParams.type });
+      await validateBulkEditRule({
+        mlAuthz,
+        ruleType: ruleParams.type,
+        edit: actions,
+        immutable: ruleParams.immutable,
+      });
       return ruleParamsModifier(ruleParams, paramsActions);
     },
   });
+
+  // rulesClient bulkEdit currently doesn't support bulk mute/unmute.
+  // this is a workaround to mitigate this,
+  // until https://github.com/elastic/kibana/pull/138900 is resolved
+  // if rule actions has been applied:
+  // - we go through each rule
+  // - mute/unmute if needed, refetch rule
+  // calling mute for rule needed only when rule was unmuted before and throttle value is  NOTIFICATION_THROTTLE_NO_ACTIONS
+  // calling unmute needed only if rule was muted and throttle value is not NOTIFICATION_THROTTLE_NO_ACTIONS
+  const rulesAction = attributesActions
+    .reverse()
+    .find(({ type }) =>
+      [BulkActionEditType.set_rule_actions, BulkActionEditType.add_rule_actions].includes(type)
+    ) as bulkActionEditPayloadRuleActions;
+
+  if (rulesAction) {
+    const errors: BulkEditError[] = [];
+    const rules = await pMap(
```

Comment:
Just trying to understand the use of `pMap` here. My understanding from the codebase is that we need to respect the `MAX_RULES_TO_UPDATE_IN_PARALLEL` to prevent performance issues and timeouts.

However, I've seen the elsewhere in the code that the `initPromisePool` method from `x-pack/plugins/security_solution/server/utils/promise_pool.ts` is used instead of `pMap`. Any reason not use use that instead?

Reply:
If I remember correctly `initPromisePool` was introduced to make efficient calls in parallel, i.e. what is pMap doing.
pMap is used widely across Kibana and even in Security Solution, thus my choice of using it.
initPromisePool also have some additional logic of wrapping executed callbacks and returning them in a certain format, which is not needed for my case.

I guess, eventually we would like to replace initPromisePool, if there is no drawback of using pMap where can be used.